### PR TITLE
tests: add integration test for billing limits

### PR DESCRIPTION
**Pull request by xvello**
- Add a `redis_key_prefix` config and implement it in our redis wrapper
- Add `EphemeralRedis` to our integration tests helpers, deletes all subkeys on Drop
- Add `it_enforces_billing_limits` integration test

### CODE REVIEW

File: capture-server/tests/events.rs [hunk 1]
```diff
@@ -133,13 +134,12 @@ async fn it_does_not_partition_limit_different_ids() -> Result<()> {
...
 
     let topic = EphemeralTopic::new().await;
-
-    let mut config = DEFAULT_CONFIG.clone();
-    config.kafka.kafka_topic = topic.topic_name().to_string();
-    config.burst_limit = NonZeroU32::new(1).unwrap();
-    config.per_second_limit = NonZeroU32::new(1).unwrap();
-
-    let server = ServerHandle::for_config(config);
+    let server = ServerConfig::new(|config| {
```
Comment (ellie) [inline capture-server/tests/events.rs:137]: I think this `mutator` is making things pretty unclear

It's probably worth considering just implementing `Default`, and/or using the builder pattern.

We can then use [typed-builder](https://github.com/idanarye/rust-typed-builder) or similar

Comment (xvello) [reply capture-server/tests/events.rs:137]: Agreed, I'm fighting against the limitations of `envconfig` right now, I'll look into switching to a config lib that allows to merge partial configs

Comment (ellie) [reply capture-server/tests/events.rs:137]: I am pretty generally pretty happy with [config-rs](https://github.com/mehcode/config-rs)

**Comment by xvello** [comment]
Should be implemented, but blocked on switching configuration libs to something that makes integration tests less clunky to write and maintain.
Would be great if you could bring the branch over to the new repo.

**Comment by bretthoerner** [comment]
Cool, here it is at the new repo: https://github.com/PostHog/hog-rs/tree/xvello/limits-itest

**Pull request closed without merging**